scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/EquationSolvingAlgorithms.java

```diff
@@ -141,8 +144,8 @@ private IteratingResult nextValueAt(
           }
         }
         cur = chooseRandomX(min, max);
-        //if min == max, another call to random will have no effect and thus we should exit
-      } while(i < maxIteration && !min.isEqualTo(max));
+        //if all timepoints have been visited or min == max, another call to random will have no effect and thus we should exit
+      } while(i < maxIters);
```

Comment:
I don't see how this leads to an infinite loop in the case you describe ? Can you elaborate either on the PR description or in an answer to that comment ?

Reply:
I came across a run that called this method with min = 4 micros and max = 6 micros, so only 4 or 5 would be randomly encountered.  At some point, the `if` statement always fails, and `i` is never incremented, so the loop never ends.

Reply:
OK! it sounds like the increment of `i` is at fault here but that's a reasonable workaround.

Reply:
`i` was only incremented for values that weren't already visited.  So, I don't think always incrementing `i` is a good fix.  It think this fix is appropriate.